src/utils/mutexStorage.ts

```diff
@@ -0,0 +1,76 @@
+import { merge } from 'lodash-es'
+
+class MutexStorage<T extends browser.storage.StorageValue> {
```

Comment:
🤔Did we make things too complex?

Reply:
Maybe... Let me test if everything works without this then I will remove it.

Reply:
Or find a npm library to do this 🤣

Reply:
It's necessary. Our use case on `browser.local.storage` API isn't thread-safe. We haven't waited for the result before visiting the newest storage.